File: src/main/kotlin/school/lec03/Lec03Main.java
package school.lec03;


public class Lec03Main {
    public static void main(String[] args) {
        Person person = new Person("asdf",15);
        Child child = new Child("asdf",18);
        printAgeIfPerson(person);
        printAgeIfPerson(child);
    }

    public static void printAgeIfPerson(Object obj){
        if(obj instanceof Person){
            Person person = (Person) obj;
            System.out.println(person.getAge());
        }
    }
}

File: src/main/kotlin/school/lec03/Lec03Main.kt
package school.lec03

fun main(){

}
fun printAgeIfPerson(obj: Any){
    if(obj is Person){
        val person = obj
        println(person.age)
    }
}

fun printAgeIfPersonNullable(obj: Any?){
    val person = obj as? Person
    print(person?.age)
}